src/main/java/edu/hm/hafner/analysis/parser/PolyspaceParser.java

```diff
@@ -0,0 +1,102 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.LookaheadParser;
+import edu.hm.hafner.analysis.Severity;
+import edu.hm.hafner.util.LookaheadStream;
+
+import static edu.hm.hafner.util.IntegerParser.*;
+
+/**
+ * A parser for Polyspace Bug Finder and Code Prover results.
+ *
+ * @author Eva Habeeb
+ */
+public class PolyspaceParser extends LookaheadParser {
+
+    private static final long serialVersionUID = -1251248150596418714L;
+
+    private static final String WARNING_PATTERN = "^([0-9]+)\\t([^\\t]*)\\t([^\\t]*)\\t([^\\t]*)\\t([^\\t]*)"
+            + "\\t([^\\t]*)\\t([^\\t]*)\\t([^\\t]*)\\t([^\\t]*)\\t([^\\t]*)\\t([^\\t]*)\\t([^\\t]*)\\t([^\\t]*)"
+            + "\\t([^\\t]*)\\t([^\\t]*)(\\t([^\\t]*))?";
```

Comment:
From my understanding it does not make sense to parse those files with regular expressions. You should directly derive from `IssueParser` and use a String splitter to get the elements you need.

Reply:
This also would help to eliminate the `null` path errors, since a matcher might return `null`, the splitter not.

Reply:
Done. Used IssueParser instead of LookaheadParser using the parse function with readStream function. 
Used String splitter instead of regular expressions.